src/devices/vdml_link.c

```diff
@@ -0,0 +1,234 @@
+/**
+ * \file vdml_link.c
+ *
+ * \brief Contains source code for functions related to the robot to robot communications.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/api/c/link.html to learn
+ * more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2022, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+ #include "pros/link.h"
+ #include "kapi.h"
+ #include "pros/apix.h"
+ #include "vdml/vdml.h"
+ #include <string.h>
+
+#define FIFO_SIZE           512 // VEX Defined TX/RX Queue Size
+#define PROTOCOL_SIZE         4 // Protocol Size 
+
+static uint8_t start_byte = 0x33;
+
+// internal function for clearing the rx buffer 
+uint32_t _clear_rx_buf(v5_smart_device_s_t* device) {
+    uint8_t buf[FIFO_SIZE];
+    return vexDeviceGenericRadioReceive(device->device_info, 
+    (uint8_t*)buf, 
+    vexDeviceGenericRadioReceiveAvail(device->device_info));
+}
+
+uint32_t link_init(uint8_t port, char* link_id, link_type_e_t type) {
+    // not used because it's a generic serial device 
+    // claim_port_i(port - 1, E_DEVICE_RADIO);
+	if (!VALIDATE_PORT_NO(port - 1)) {
+		errno = EINVAL;
+		return PROS_ERR;
+	}
+    // by default, the vexlink radio in the lower port should be by default E_DEVICE_NONE
+    // or, the only radio on the brain
+    v5_device_e_t plugged_device = registry_get_plugged_type(port - 1);
+    if(plugged_device != E_DEVICE_NONE && plugged_device != E_DEVICE_RADIO) {
+        return PROS_ERR;
+    }
+	v5_smart_device_s_t* device = registry_get_device(port - 1);
+	if (!port_mutex_take(port - 1)) {
+		errno = EACCES;
+		return PROS_ERR;
+	}
+    vexDeviceGenericRadioConnection(device->device_info, link_id, type, false);
+    return_port(port - 1, 1);
+}
+
+uint32_t link_init_override(uint8_t port, char* link_id, link_type_e_t type) {
```

Comment:
Yeah, the reasoning behind link_init and link_init_override being 2 seperate functions was because link_init_override will be relatively niche in terms of its usage, so link_init is shorter and more convenient for most use cases.

Reply:
You can keep the api separate if you want, but the implementation can be shared.